server/src/models/Room.ts

```diff
@@ -27,12 +27,23 @@ class Room {
   }
 
   addUser(user: User): void {
-    const name = user.name;
+    let userName = user.name;
+    let numberUser = 0;
 
-    if (this.users.find(user => user.name === name)) throw new Error('This user nam already exists in room');
+    while (this.isTaken(userName)) {
+      numberUser++;
+      userName = user.name + `(${numberUser})`;

```

Comment:
tu możesz pominąć backticki, ewentualnie całe wyrażenie ubrać (gdybyś chciał to ubrać w nawiasy albo dodać spacje) dla ładniejszego efektu :)